src/Http/Http.Extensions/gen/RequestDelegateGeneratorSources.cs

```diff
@@ -449,6 +449,27 @@ public override bool IsDefined(Type attributeType, bool inherit)
     }
 """;
 
+    public static string ParameterBindingMetadataClass = $$"""
+    {{GeneratedCodeAttribute}}
+    file sealed class ParameterBindingMetadata(
+        string name,
+        ParameterInfo parameterInfo,
+        bool hasTryParse = false,
+        bool hasBindAsync = false,
+        bool isOptional = false) : Microsoft.AspNetCore.Http.Metadata.IParameterBindingMetadata
```

Comment:
Does using primary constructors in the generated code force the user to use C# 12? Is that a concern at all?

Reply:
Good call! I believe we had some discussions around this in .NET 8 and concluded that we didn't want to tie ourselves to C#12 for this changeset.